src/pykeen/models/nbase.py

```diff
@@ -0,0 +1,475 @@
+    as well as a matching instance of a :class:`pykeen.nn.Interaction`.
+    """
+
+    #: The default regularizer class
+    regularizer_default: ClassVar[Optional[Type[Regularizer]]] = None
+    #: The default parameters for the default regularizer class
+    regularizer_default_kwargs: ClassVar[Optional[Mapping[str, Any]]] = None
+
+    def _reset_parameters_(self):  # noqa: D401
+        """Reset all parameters of the model in-place."""
+        # cf. https://github.com/mberr/ea-sota-comparison/blob/6debd076f93a329753d819ff4d01567a23053720/src/kgm/utils/torch_utils.py#L317-L372   # noqa:E501
+        # Make sure that all modules with parameters do have a reset_parameters method.
+        uninitialized_parameters = set(map(id, self.parameters()))
+        parents = defaultdict(list)
+
+        # Recursively visit all sub-modules
+        task_list = []
+        for name, module in self.named_modules():
+            # skip self
+            if module is self:
+                continue
+
+            # Track parents for blaming
+            for p in module.parameters():
+                parents[id(p)].append(module)
+
+            # call reset_parameters if possible
+            if hasattr(module, 'reset_parameters'):
+                task_list.append((name.count('.'), module))
+
+        # initialize from bottom to top
+        # This ensures that specialized initializations will take priority over the default ones of its components.
+        for module in map(itemgetter(1), sorted(task_list, reverse=True, key=itemgetter(0))):
+            module.reset_parameters()
+            uninitialized_parameters.difference_update(map(id, module.parameters()))
+
+        # emit warning if there where parameters which were not initialised by reset_parameters.
+        if len(uninitialized_parameters) > 0:
+            logger.warning(
+                'reset_parameters() not found for all modules containing parameters. '
+                '%d parameters where likely not initialized.',
+                len(uninitialized_parameters),
+            )
+
+            # Additional debug information
+            for i, p_id in enumerate(uninitialized_parameters, start=1):
+                logger.debug('[%3d] Parents to blame: %s', i, parents.get(p_id))
+
+    def _instantiate_default_regularizer(self, **kwargs) -> Optional[Regularizer]:
+        """Instantiate the regularizer from this class's default settings.
+
+        If the default regularizer is None, None is returned.
+        Handles the corner case when the default regularizer's keyword arguments are None
+        Additional keyword arguments can be passed through to the `__init__()` function
+        """
+        if self.regularizer_default is None:
+            return None
+
+        _kwargs = dict(self.regularizer_default_kwargs or {})
+        _kwargs.update(kwargs)
+        return self.regularizer_default(**_kwargs)
+
+    def post_parameter_update(self) -> None:
```

Comment:
This is called in the training loops. Hence each `Model` needs to implement this method. Should we move it to the base class?

Reply:
Good call. Done in 7d4a4b59